### PR TITLE
[WIP] add command for migrating views

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
add a command for migrating views
factorize code

### Linked issues
Resolves https://github.com/databrickslabs/ucx/issues/1172

### Functionality 

- [ ] added relevant user documentation
- [x] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests
- [ ] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

Missing integration tests
Depends on #1177

### CODE REVIEW

**Review by qziyuan — CHANGES_REQUESTED** [review]
Hi @ericvergnaud could you take a look at and double check this issue #1273? I think the old `_migrate_view_table` function is creating UC view with old HMS dependencies.

**Comment by ericvergnaud** [comment]
@qziyuan sure - I've asked @nfx how I should prioritize

File: src/databricks/labs/ucx/runtime.py [hunk 1]
```diff
@@ -433,27 +433,42 @@ def migrate_external_tables_sync(
...
     udf_crawler = UdfsCrawler(sql_backend, cfg.inventory_database)
     grant_crawler = GrantsCrawler(table_crawler, udf_crawler)
     table_mappings = TableMapping(install, ws, sql_backend)
     migration_status_refresher = MigrationStatusRefresher(ws, sql_backend, cfg.inventory_database, table_crawler)
     group_manager = GroupManager(sql_backend, ws, cfg.inventory_database)
     mount_crawler = Mounts(sql_backend, ws, cfg.inventory_database)
-    cluster_locations = {}
-    if ws.config.is_azure:
-        locations = ExternalLocations(ws, sql_backend, cfg.inventory_database)
-        azure_client = AzureAPIClient(
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/runtime.py:446]: See message in Slack from @nfx about these calls only existing from the cli not within a job

**Review by JCZuurmond — CHANGES_REQUESTED** [review]
Your changes LGTM. However, there was a problem already that should be addressed first, see the comment above.

**Comment by nfx** [comment]
closing in favor of https://github.com/databrickslabs/ucx/pull/1325

**Pull request closed without merging**